componentLibraries/defaultLibrary/Signal/Logic/SignalSteadyStateIdentifier.hpp

```diff
@@ -0,0 +1,217 @@
+
+ Copyright 2017 Hopsan Group
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+
+ The full license is available in the file LICENSE.
+ For details about the 'Hopsan Group' or information about Authors and
+ Contributors see the HOPSANGROUP and AUTHORS files that are located in
+ the Hopsan source code root directory.
+
+-----------------------------------------------------------------------------*/
+
+//!
+//! @file   SignalSteadyStateIdentifier.hpp
+//! @author Robert Braun <robert.braun@liu.se>
+//! @date   2022-03-10
+//!
+//! @brief Steady-state identifier component
+//!
+//$Id$
+
+#ifndef SIGNALSTEADYSTATEIDENTIFIER_H
+#define SIGNALSTEADYSTATEIDENTIFIER_H
+
+#include "ComponentEssentials.h"
+#include "ComponentUtilities.h"
+
+namespace hopsan {
+
+    //!
+    //! @brief
+    //! @ingroup SignalComponents
+    //!
+    class SignalSteadyStateIdentifier : public ComponentSignal
+    {
+
+    private:
+        double *mpIn, *mpOut, *mpTol, *mpWl, *mpSd, *mpL1, *mpL2, *mpL3;
+        int mMethod;
+        std::vector<double> mWindow;
+        size_t mWindowId;
+        double mDelayedX;
+        double mDelayedXf;
+        double mDelayedDf;
+
+    public:
+        static Component *Creator()
+        {
+            return new SignalSteadyStateIdentifier();
+        }
+
+        void configure()
+        {
+            addInputVariable("in", "", "", 0.0, &mpIn);
+            addOutputVariable("out", "1 if steadystate, else 0", "", &mpOut);
+
+            std::vector<HString> methods;
+            methods.push_back("1: Rectangular sliding window");
+            methods.push_back("2: Variance ratio test (F-test)");
+            methods.push_back("3: Exponentially weighted moving average (R-test)");
+
+            addConditionalConstant("method", "Estimation method", methods, 0, mMethod);
+
+            addInputVariable("tol", "Tolerance", "", 1.0, &mpTol);
+            addInputVariable("wl", "Length of sliding window (for methods 1 and 2)", "", 0.1, &mpWl);
+            addInputVariable("sd", "Standard deviation of white noise (for methods 2 and 3)", "", 0.01, &mpSd);
+            addInputVariable("l1", "Filter factor 1 (for method 3)", "", 0.1, &mpL1);
+            addInputVariable("l2", "Filter factor 2 (for method 3)", "", 0.1, &mpL2);
+            addInputVariable("l3", "Filter factor 3 (for method 3)", "", 0.1, &mpL3);
+        }
+
+
+        void initialize()
+        {
+            mWindowId = 0;
```

Comment:
Does this mean that the window always begins from the beginning

Reply:
This only means that we start on index 0 in the circular buffer (see below).